ihwang/42sh
# Dudu/Reading history from history file

This PR is for improving the read history from the history file
**What it does?**
- It reads from the history file which include the ctrl-D character, normal command, command has quotation and put them into correct order.

**How to test?**
 + Step 1: Make combination of command with normal command, command with quotation and command which terminated with ctrl-D
 Make the history size smaller and larger than 50 lines of the commands:
 Notice: If you make more than 50 commands, and exit, it will save only the first 50 commands, so pls test throughly
+ Step 2: Exit from the shell
+ Step3: go to the shell again
+ Step4: put command $> fc -l
+ Step5: check if there is any weird from the history, like a line without a command, or 2 commands next to each other, ...

**Is there any limitation**
Yes, there is a limitation when mixing ctrl-d, quotation, and inhibitor. However, try not to do that, since it is a complex problem

**Acceptance criteria**
+ Adding into the history container correctly with ctrl-d (quotation or with logical and pipe operator)
+ Adding into the history container correctly with command lines
+ Adding into the history container correctly with command lines including quotation
